oci/layout/oci_dest.go

```diff
@@ -299,6 +304,102 @@ func (d *ociImageDestination) CommitWithOptions(ctx context.Context, options pri
 	return os.WriteFile(d.ref.indexPath(), indexJSON, 0644)
 }
 
+// tryReflinkLocalFile attempts to reflink the specified file and digest it.
+// If relinking does not work, reset to doing a verbatim copy of the file.
+func tryReflinkLocalFile(dest *ociImageDestination, file string) (private.UploadedBlob, bool, error) {
+	fInfo, err := os.Stat(file)
+	if err != nil {
+		return private.UploadedBlob{}, false, err
+	}
+
+	blobFile, err := os.CreateTemp(dest.ref.dir, "oci-put-blob")
+	if err != nil {
+		return private.UploadedBlob{}, false, err
+	}
+	blobName := blobFile.Name()
+
+	copyRange := false
```

Comment:
Is there a reason we would not want to use this?

Reply:
It turns out that Go (in `go/src/os/zero_copy_linux.go`) automatically uses this within an ordinary `io.Copy` between two `*os.File` values, this does not need to be special-cased.